lc-hax/Scripts/Core/HaxCamera.cs

```diff
@@ -0,0 +1,178 @@
+
+
+        this.MousePan.enabled = true;
+        this.KeyboardMovement.enabled = true;
+        this.HaxCamAudioListener.enabled = true;
+        this.HaxCamContainer.SetActive(false);
+        this.CustomCamera = newCam;
+        DontDestroyOnLoad(this.HaxCamContainer);
+        return newCam;
+    }
+
+
+    internal void CopyFromCamera(Transform container, ref Camera Cam, ref Camera camData) {
+        Cam.transform.position = Vector3.zero;
+        Cam.transform.rotation = Quaternion.identity;
+        Cam.transform.localPosition = Vector3.zero;
+        Cam.transform.localRotation = Quaternion.identity;
+        Cam.transform.localScale = Vector3.one;
+
+        // Copy camera settings
+        Cam.CopyFrom(camData);
+
+        // Ensure the custom camera has the same culling mask as the original camera
+        Cam.cullingMask = camData.cullingMask;
+
+        // Copy other relevant properties
+        Cam.clearFlags = camData.clearFlags;
+        Cam.backgroundColor = camData.backgroundColor;
+        Cam.nearClipPlane = camData.nearClipPlane;
+        Cam.farClipPlane = camData.farClipPlane;
+        Cam.fieldOfView = camData.fieldOfView;
+        Cam.depth = camData.depth;
+        Cam.renderingPath = camData.renderingPath;
+
+        // Make it work as the actual camera
+        Cam.tag = camData.tag;
+
+        // Get or add the HDAdditionalCameraData component
+        HDAdditionalCameraData? hdData = Cam.GetComponent<HDAdditionalCameraData>();
+        hdData ??= Cam.gameObject.AddComponent<HDAdditionalCameraData>();
+
+        if (camData.TryGetComponent(out HDAdditionalCameraData dataToCopy)) {
+            if (hdData != null && dataToCopy != null) {
+                hdData.customRenderingSettings = true;
+                hdData.renderingPathCustomFrameSettingsOverrideMask.mask =
+                    dataToCopy.renderingPathCustomFrameSettingsOverrideMask.mask;
+                hdData.renderingPathCustomFrameSettings.SetEnabled(FrameSettingsField.CustomPass,
+                    dataToCopy.renderingPathCustomFrameSettings.IsEnabled(FrameSettingsField.CustomPass));
+                hdData.renderingPathCustomFrameSettings.SetEnabled(FrameSettingsField.Volumetrics,
+                    dataToCopy.renderingPathCustomFrameSettings.IsEnabled(FrameSettingsField.Volumetrics));
+                hdData.renderingPathCustomFrameSettings.lodBiasMode =
+                    dataToCopy.renderingPathCustomFrameSettings.lodBiasMode;
+                hdData.renderingPathCustomFrameSettings.lodBias = dataToCopy.renderingPathCustomFrameSettings.lodBias;
+                hdData.antialiasing = dataToCopy.antialiasing;
+                hdData.renderingPathCustomFrameSettings.SetEnabled(FrameSettingsField.ShadowMaps,
+                    dataToCopy.renderingPathCustomFrameSettings.IsEnabled(FrameSettingsField.ShadowMaps));
+            }
+        }
+        else {
+            Destroy(hdData);
```

Comment:
Why do we need to destroy?

Reply:
Because if there's no reason to leave a un set component if the original camera does not have it.

Reply:
Then why not check if the original camera has it first and if it doesn't don't attempt to get or add the component to the new camera?

Reply:
Also, I am curious when will HDAdditionalCameraData not be present?

Reply:
> Also, I am curious - when will HDAdditionalCameraData not be present?

I inspected a mod and the vanilla’s camera does not have it, this mod adds and sets these values, which this way we can preserve what quality the camera is set to

Reply:
Why does our mod add it?

Reply:
> Why does our mod add it? Or are you saying other mods remove it?

Our mod that way supports any additional hd parameters such as hdlethalcompany sets for all cameras using this component

Reply:
then isn't this sufficient?

```cs
        if (originalCamera.TryGetComponent(out HDAdditionalCameraData dataToCopy)) {
            HDAdditionalCameraData? hdData =
                camera.GetComponent<HDAdditionalCameraData>() ??
                camera.gameObject.AddComponent<HDAdditionalCameraData>();

            hdData.customRenderingSettings = true;
            hdData.renderingPathCustomFrameSettingsOverrideMask.mask = dataToCopy.renderingPathCustomFrameSettingsOverrideMask.mask;
            hdData.renderingPathCustomFrameSettings.lodBiasMode = dataToCopy.renderingPathCustomFrameSettings.lodBiasMode;
            hdData.renderingPathCustomFrameSettings.lodBias = dataToCopy.renderingPathCustomFrameSettings.lodBias;
            hdData.antialiasing = dataToCopy.antialiasing;

            hdData.renderingPathCustomFrameSettings.SetEnabled(
                FrameSettingsField.CustomPass,
                dataToCopy.renderingPathCustomFrameSettings.IsEnabled(FrameSettingsField.CustomPass)
            );

            hdData.renderingPathCustomFrameSettings.SetEnabled(
                FrameSettingsField.Volumetrics,
                dataToCopy.renderingPathCustomFrameSettings.IsEnabled(FrameSettingsField.Volumetrics)
            );

            hdData.renderingPathCustomFrameSettings.SetEnabled(
                FrameSettingsField.ShadowMaps,
                dataToCopy.renderingPathCustomFrameSettings.IsEnabled(FrameSettingsField.ShadowMaps)
            );
        }
 ```